test/testBopomofoUtil.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * chewing-editor: Chewing userphrase editor
+ * Copyright (C) 2014 Chewing Development Team
+
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "gtest/gtest.h"
+
+#include "BopomofoUtil.hpp"
+
+#if (defined _MSC_VER) && (_MSC_VER >= 1600)
+#    pragma execution_character_set("utf-8")
+#endif
+
+class TestBopomofoUtil : public ::testing::Test {
+protected:
+    TestBopomofoUtil() = default;
+    virtual ~TestBopomofoUtil() = default;
+};
+
+TEST_F(TestBopomofoUtil, ReduceDuplicatedSpacesInsideBopomofo)
+{
+    ASSERT_EQ(0, QString::compare("ㄘㄜˋ ㄕˋ", BopomofoUtil::normalize("ㄘㄜˋ  ㄕˋ")));
+    ASSERT_EQ(0, QString::compare("ㄘㄜˋ ㄕˋ", BopomofoUtil::normalize("ㄘㄜˋ   ㄕˋ")));
```

Comment:
Nop, this PR is just moving the original part of replacing ambugity words into a sperated file and adding the reducing duplicated spaces function.

But, yes, we still need tests for replacing ambugity words, that's why I move it. After this PR, I'll submit another PR for the tests.